core/src/main/java/com/graphhopper/routing/OSMReaderConfig.java

```diff
@@ -28,6 +32,54 @@ public class OSMReaderConfig {
     private double longEdgeSamplingDistance = Double.MAX_VALUE;
     private int workerThreads = 2;
 
+    public static List<String> createDefaultAcceptedHighways() {
+        return Arrays.asList(
+                "motorway", "motorway_link",
+                "trunk", "trunk_link",
+                "primary", "primary_link",
+                "secondary", "secondary_link",
+                "tertiary", "tertiary_link",
+                "unclassified",
+                "residential",
+                "living_street",
+                "service",
+                "pedestrian",
+                "track",
+                "bus_guideway", "busway",
+                "escape", "emergency_bay",
+                "raceway",
+                "road",
+                "footway",
+                "bridleway",
+                "steps",
+                "corridor",
+                "path",
+                "cycleway",
+                "proposed",
+                "construction"
+        );
+    }
+
+    public List<String> getAcceptedHighways() {
+        return acceptedHighways;
+    }
+
+    /**
+     * Sets the values of the highway tag that shall be considered when we read the OSM file. OSM ways that do not include
+     * a highway tag with one of these values will be ignored entirely during import, but just because they are
+     * accepted does not necessarily mean they will be considered by the route calculations. This still depends on the
+     * weighting and/or the access flags for the corresponding edges. Not including some road types here is mostly a
+     * performance optimization. For example if one is only interested in routing for motorized vehicles the routing
+     * graph size can be reduced by not including tracks, footways and paths (20-50% depending on your area, ~25%
+     * for planet files). Another reason to exclude footways etc. for motorized vehicle routing could be preventing
+     * undesired u-turns (#1858). Similarly, one could exclude motorway, trunk or even primary for bicycle or pedestrian routing, but since there
+     * aren't many such roads the graph size reduction will be very small (<4%).
```

Comment:
I tested this for Germany where excluding footways,tracks and paths reduced the number of edges by around 50%. I think this is simply because in Germany many such ways are actually mapped. Looking at taginfo the reduction should be more around 25% for the planet file, because in most countries not all of these smaller ways are mapped (and probably in many areas not so many such roads exist).